### PR TITLE
Refactor/walk objects

### PR DESCRIPTION
**Pull request by toaster**
This PR refactors glCanvas.walkObjects.
The refactoring is a preparation for moving the walk method out of the driver package.

### CODE REVIEW

File: driver/gl/window.go [hunk 1]
```diff
@@ -488,26 +488,28 @@ func (w *window) findObjectAtPositionMatching(canvas *glCanvas, mouse fyne.Posit
...
 		y2 := pos.Y + walked.Size().Height
 		if mouse.X >= x2 || mouse.Y >= y2 {
-			return
+			return false
 		}
 
 		if matches(walked) {
 			found = walked
 			foundX, foundY = mouse.X-pos.X, mouse.Y-pos.Y
+			return true
```
Comment (andydotxyz) [inline driver/gl/window.go]: This return true may not be correct... We actually need to find the deepest object in a tree that matches the criteria (such as a Tappable object within a Tappable object, or Scrollable within Scrollable). Which means sometimes we must pass a found object to found a more relevant result.

File: driver/gl/draw.go [hunk 2]
```diff
@@ -9,66 +9,52 @@ import (
...
+// - fAfterChildren is called for the obj after traversing the obj's children
+// The walk can be aborted by returning true in one of the functions:
+// - if fBeforeChildren returns true, further traversing is stopped immediatly, the after function
+//   will not be called for the obj where the walk stopped, however, it will be called for all its
+//   parents
+// - if a walk has been stopped, the after function is called with the third argument set to true
+func (c *glCanvas) walkObjects(
+	obj fyne.CanvasObject,
+	pos fyne.Position,
+	fBeforeChildren func(fyne.CanvasObject, fyne.Position) bool,
```
Comment (andydotxyz) [inline driver/gl/draw.go]: We don't use type prefix for variables, so perhaps beforeChildren or callbackBefore may be more in-keeping?

File: driver/gl/window.go (hunk 1 above)
Comment (toaster) [reply driver/gl/window.go]: Oh … I will try to create a test case but I'm not sure that I will succeed. However, the `true` will be changed into `false`.

**Comment by toaster** [comment]
integrated feedback and merged current develop branch

**Pull request merged**